### PR TITLE
doc: optimizing subpage levels in App development section

### PR DESCRIPTION
**Pull request by peknis**
Reorganizing the documentation of the Application
development section.

Ref: NCSDK-12679

Signed-off-by: Pekka Niskanen <pekka.niskanen@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ext_comps.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.. _ext_components:
+
+Using external components
+#########################
+
+The following user guides show how you can use external components in your application:
```
Comment (greg-fer) [inline doc/nrf/ext_comps.rst:6]:
```suggestion
The following user guides show how you can use external components in your application.
```
There's the Subpages header after the colon, so I'd suggest a period here. We don't introduce headers with colons.

File: doc/nrf/app_opt.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.. _app_optimize:
+
+Optimizing your application
+###########################
+
+The following user guides show how you can optimize your application:
```
Comment (greg-fer) [inline doc/nrf/app_opt.rst:6]:
```suggestion
The following user guides show how you can optimize your application.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -185,6 +185,14 @@ Documentation
 
 In addition to documentation related to the changes listed above, the following documentation has been updated:
 
-* |no_changes_yet_note|
+* Reorganized the contents of the :ref:´ug_app_dev` section:
+
+  * Reduced the toc levels of the subpages.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:190]:
```suggestion
  * Reduced the ToC levels of the subpages.
```
Also in line 196.
But to be honest, in my opinion this change could be omitted, as it is on the verge of minor/trivial.

File: doc/nrf/app_opt.rst [hunk 4]
```diff
@@ -0,0 +1,13 @@
+.. _app_optimize:
+
+Optimizing your application
```
Comment (greg-fer) [inline doc/nrf/app_opt.rst:3]:
```suggestion
Optimizing application
```

**Review by greg-fer — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**